support-frontend/assets/pages/paper-subscription-checkout/helpers/options.ts

```diff
@@ -14,24 +14,43 @@ import {
 	ActivePaperProductTypes,
 	Everyday,
 } from 'helpers/productPrice/productOptions';
+import * as storage from 'helpers/storage/storage';
 import type { DateYMDString } from 'helpers/types/DateString';
 import { getQueryParameter } from 'helpers/urls/url';
 import { formatMachineDate } from 'helpers/utilities/dateConversions';
 import { getHomeDeliveryDays } from 'pages/paper-subscription-checkout/helpers/homeDeliveryDays';
 import { getVoucherDays } from 'pages/paper-subscription-checkout/helpers/voucherDeliveryDays';
 
 function getProductOption(): PaperProductOptions {
-	const productInUrl = getQueryParameter('product');
-
+	const PRODUCT_STORAGE_KEY = 'selectedProduct'; // create the key for product in session storage
+	const productInUrl = getQueryParameter('product'); // get product from Query |Params
+	const productInSessionStorage = storage.getSession(PRODUCT_STORAGE_KEY); // get product from Session Storage
+	const productInUrlOrSessionStorage = productInUrl
+		? productInUrl
+		: productInSessionStorage; // get the value from url or session storage(whichever is available)
+	if (productInUrl) {
+		storage.setSession(PRODUCT_STORAGE_KEY, productInUrl); // always set product in session storage if  there is a value in the url
+	}
 	return (
-		ActivePaperProductTypes.find((product) => product === productInUrl) ??
-		Everyday
+		ActivePaperProductTypes.find(
+			(product) => product === productInUrlOrSessionStorage,
+		) ?? Everyday
 	);
 }
 
 function getFulfilmentOption(): PaperFulfilmentOptions {
```

Comment:
Same as above, I don't think we need the changes to store the `fulfilment` in session storage anymore as the `redirectAfterSignOut` session storage technique now negates the need to do this?